src/Dapr.Client/DaprClientGrpc.cs

```diff
@@ -1373,6 +1375,537 @@ private Autogenerated.StateItem ToAutogeneratedStateItem(StateTransactionRequest
             var resp = await client.UnsubscribeConfigurationAsync(request, options);
             return new UnsubscribeConfigurationResponse(resp.Ok, resp.Message);
         }
+
+        #endregion
+
+        #region Cryptography
+
+        private async Task<byte[]> EncryptStreamAsync(string vaultResourceName, Stream stream,
+            KeyWrapAlgorithm algorithm, string keyName, DataEncryptionCipher dataEncryptionCipher,
+            string decryptionKeyName, bool omitDecryptionKey, CancellationToken cancellationToken = default)
+        {
+            ArgumentVerifier.ThrowIfNullOrEmpty(vaultResourceName, nameof(vaultResourceName));
+            ArgumentVerifier.ThrowIfNullOrEmpty(keyName, nameof(keyName));
+
+
+            var encryptRequestOptions = new Autogenerated.EncryptRequestOptions
+            {
+                ComponentName = vaultResourceName,
+                DataEncryptionCipher = dataEncryptionCipher.GetValueFromEnumMember(),
+                KeyName = keyName,
+                KeyWrapAlgorithm = algorithm.GetValueFromEnumMember(),
+                OmitDecryptionKeyName = omitDecryptionKey
+            };
+
+            if (!omitDecryptionKey)
+            {
+                ArgumentVerifier.ThrowIfNullOrEmpty(decryptionKeyName, nameof(decryptionKeyName));
+                encryptRequestOptions.DecryptionKeyName = decryptionKeyName;
+            }
+
+            var options = CreateCallOptions(headers: null, cancellationToken);
+            var duplexStream = client.EncryptAlpha1(options);
+
+            //Start with passing the metadata about the encryption request itself in the first message
+            await duplexStream.RequestStream.WriteAsync(
+                new Autogenerated.EncryptRequest { Options = encryptRequestOptions }, cancellationToken);
+
+            //Send the plaintext bytes in 4 kb blocks in subsequent messages
+            const int blockSize = 4 * 1024; // 4 kb
```

Comment:
Should the block size be an option to the encrypt method?

Reply:
This is another relic of the [discussion](https://discord.com/channels/778680217417809931/1133104673558114439/1187856221529722910) on Discord in which the recommendation was to use 2-4 KB chunks with no more than 64KB. I'm entirely in favor of using this as a default and letting the developer pick a specific value, but we should resolve first whether we'd prefer to use method overloads to do that or switch to an options block (and its various constructor overloads) instead.

Reply:
This was a factor in the suggestion for an options object; I think it reasonable to allow users to override the block size default, but I also think it would be too messy to try to do that via method overloads.

Reply:
Added to the options with some notes in the XML documentation about maxing at 64. Typed as a uint to limit any additional validation needs.

Reply:
I don't see many situations where users may need to change the chunk size personally